tests/MSBuildDeviceIntegration/Tests/DebuggingTest.cs

```diff
@@ -242,7 +242,8 @@ public override void OnCreate ()
 					int port = rnd.Next (10000, 20000);
 					TestContext.Out.WriteLine ($"{port}");
 					var args = new SoftDebuggerConnectArgs ("", IPAddress.Loopback, port) {
-						MaxConnectionAttempts = 2000, // we need a long delay here to get a reliable connection
+						MaxConnectionAttempts = 210, // we need a long delay here to get a reliable connection
+						TimeBetweenConnectionAttempts = 2000,
```

Comment:
Is it really going to try to connect 210 times and wait 2 seconds between each?

Maybe we could put these numbers in a const, as they are used below.

Reply:
The previous settings would attempt to connect 2000 times with a 500ms delay. I'm not positive that we've failed here specifically historically, but I did see some 16-18 minute execution times for this test and this seems like the only thing that could be waiting that long. If we do fail here in the future at least we can fail and retry more quickly. 

Should we reduce the retry time further? I wanted to account for an emulator that is potentially running _very_ slowly, but maybe 3-5 minutes would be sufficient here rather than 7?

Reply:
I dropped this down to 100 attempts with a 3 second wait for a total of 5 minutes.